src/components/NavbarLink/index.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import './styles.css'
+import { useHistory } from 'react-router-dom'
+import PropTypes from 'prop-types'
+
+const NavbarLink = ({ path, text, iconSrc }) => {
+  const history = useHistory()
+
+  return (
+    <button onClick={() => history.push(path)} className='link'>
+      {text || null}
+      {iconSrc ? <img src={iconSrc} alt='Imagen' /> : null}
```

Comment:
The `nulls` are not necessary. You can use `{text}` directly (if you never pass a text, then it renders nothing)

For the second one it's ok, you can write `{iconSrc && <img. ....}` instead. if you pass an iconSrc source then you render the image, otherwise nothing gets rendered.

[Conditional rendering](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)